waterwheel/wheel_util.py

```diff
@@ -0,0 +1,189 @@
+import re
+import os
+import pickle
+import json
+import pandas as pd
+from pathlib import Path
+from typing import Dict, List
+from tqdm import tqdm
+
+from spacy.tokens import Doc, Span, DocBin
+from spacy.matcher import PhraseMatcher
+
+doc_bin_file = Path(os.path.dirname(os.path.realpath(__file__))) / 'resources/doc_bin.pkl'
+vocab_file = Path(os.path.dirname(os.path.realpath(__file__))) / 'resources/vocab.json'
+wikidata_file = Path(os.path.dirname(os.path.realpath(__file__))) / 'resources/wikidata.json'
+stop_words_file = Path(os.path.dirname(os.path.realpath(__file__))) / 'resources/stop_words.txt'
+
+def read_files(nlp):
+    """Load necessary data from resource files.
+    
+    Parameters
+    ----------
+    nlp:
+        spacy nlp object
+    
+    Returns
+    -------
+    vocab: Dict
+        A dictionary containing the different water body types and their hash value.
+    wikidata: Dict
+        A dictionary with each wikilink data for every water_body
+    stop_words: List
+        A set of common words in English
+    matcher: PhraseMatcher
+        A PhraseMatcher object with all vocab loaded.
+    """
+    
+    vocab = {}
+    wikidata = {}
+    stop_words = set()
+    matcher = None
+    
+    try:
+        with open(vocab_file) as file:
+            vocab = json.load(file)
+    except Exception as e:
+        raise RuntimeError(f'Failed to load vocab from {vocab_file}.\nError: {str(e)}')
+
+    try:
+        with open(wikidata_file) as file:
+            wikidata = json.load(file)
+    except Exception as e:
+        raise RuntimeError(f'Failed to load wikidata from {wikidata_file}.\nError: {str(e)}')
+
+    try:
+        with open(stop_words_file) as file:
+            stop_words = set(file.read().splitlines())
+    except Exception as e:
+        raise RuntimeError(f'Failed to load stop_words from {stop_words_file}.\nError: {str(e)}')
+
+    try:
+        with open(doc_bin_file, 'rb') as file:
+            doc_bins_bytes = pickle.load(file)
+        doc_bins = {key: DocBin().from_bytes(value) for key, value in doc_bins_bytes.items()}
+        phrases_bin = {key: list(bin.get_docs(nlp.vocab)) for key, bin in doc_bins.items()}
+        matcher = PhraseMatcher(nlp.vocab, attr='LOWER')
+        for key, phrases in phrases_bin.items():
+            matcher.add(key.upper(), None, *phrases)
+            #matcher.add(key.upper(), phrases)
+    except Exception as e:
+        raise RuntimeError(f'Failed to load water bodies from {doc_bin_file_file}.\nError: {str(e)}')
+    
+    return vocab, wikidata, stop_words, matcher
+
+def write_files(vocab, wikidata, phrase_map):
```

Comment:
Shouldn't these methods be in scripts? Data prep shouldn't be part of Waterwheel itself?

Reply:
This file offers functionality to append the existing vocab. So if someone wanted to add to (or replace) the existing base of all the water bodies they can do so using the module. I thought this could be useful if someone had a very specific set of rivers/lakes etc to match. So I kept it as part of the package. 
If we don't want to distribute this extension as of now then I can degrade it normal script files outside of the package.

Reply:
I don't think we want to support this extension right now. If we expose it, we'll have to manage it. Better to be conservative at the outset and open up functionality gradually over time.

Reply:
`+1` on keeping the surface area of exposed API small

Reply:
Done.